data_structures/binary_tree/segment_tree_other.py

```diff
@@ -187,7 +220,47 @@ def _update_tree(self, node, i, val):
             self._update_tree(node.right, i, val)
         node.val = self.fn(node.left.val, node.right.val)
 
+    def _update_tree_range(
+        self,
+        node: SegmentTreeNode,
+        i: int,
+        j: int,
+        val: T,
+    ) -> None:
+
```

Comment:
A function with no doctests?

Reply:
It seems, there is no need for doctests for the `__init__` method, since one is quite straightforward - just simple assignments and no complex logic. But if they _are_ required, I can surely add them, it's not a problem.